### DISCUSSION TITLE
v16 Upgrade: Rollup unable to identify typescript path aliases and reference bundled outputs

### DISCUSSION THREAD

**Discussion by arecvlohe** [nrwl/nx#17605] [General]
I am working on upgrading from v15 to v16. The main issue I run into when upgrading is, I think, Rollup’s inability to resolve typescript path aliases. Previously this worked without any issue and all my libs could be imported correctly across apps and other libs in the monorepo. However, instead I get, for example:

```
'@artists/sf-fp' is imported by libs/header/src/lib/components/announcements.tsx, but could not be resolved – treating it as an external dependency
```

These libs are also private, we don’t publish them, so the only way we are importing them is through bundling them. They are not an external dependency now and should instead be referenced via their bundled output. If you need anymore context, let me know.

**Comment by bigcakes** [comment]
I am seeing this same thing trying to create a new publishable library with rollup to bundle up our internal library dependencies.

**Comment by thomassimko** [comment]
This is happening to me with pre-existing publishable libraries after upgrading to version 16.

**Comment by EthanML** [comment]
Same issue here. 👎  We're new users of Nx so started on v16 and want to use Rollup to bundle our published library while having it include bundled output from our other internal libraries.

**Comment by EthanML** [comment]
Anybody know of an alternative to Rollup on Nx16 for bundling a publishable lib in the meantime? I ended up trying Rollup in the first place because of the `esbuild` Nx plugin's inability to generate declaration files 😕
